src/main/java/core/basesyntax/service/serviceimpl/FruitRawStringParserImpl.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax.service.serviceimpl;
+
+import core.basesyntax.dto.FruitTransactionDto;
+import core.basesyntax.exception.DataFileCorrupted;
+import core.basesyntax.service.interfaces.FruitRawStringParser;
+import core.basesyntax.service.interfaces.FruitStringsCheck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FruitRawStringParserImpl implements FruitRawStringParser {
+    private static final int OPERATION_NEXT_INDEX = 0;
+    private static final int FRUIT_NAME_INDEX = 1;
+    private static final int FIRST_CVS_LINE = 1;
+    private static final int FRUIT_QUANTITY_INDEX = 2;
+
+    private final FruitStringsCheck checkFruitsInFile = new FruitStringsCheckImpl();
+
+    @Override
+    public List<FruitTransactionDto> parsedFruitData(List<String> rawStrings) {
+        if (!checkFruitsInFile.checkFruitQuantity(rawStrings)) {
+            throw new DataFileCorrupted("Some data in file is missing");
+        }
+        var fruitDto = new ArrayList<FruitTransactionDto>(rawStrings.size() - FIRST_CVS_LINE);
```

Comment:
this is a list of transactions, not a fruit dto
```suggestion
        var transactions = new ArrayList<FruitTransactionDto>(rawStrings.size() - FIRST_CVS_LINE);
```